runware/types.py

```diff
@@ -142,142 +142,63 @@ class IRefiner:
+    guideImageUuid: str
 
 
 @dataclass
-class IControlNetCannyWithUUID(IControlNetBaseWithUUID):
-    weight: Optional[float]
-    start_step: Optional[int]
-    end_step: Optional[int]
-    control_mode: Optional[EControlMode]
-    low_threshold_canny: int
-    high_threshold_canny: int
-    guide_image: Optional[Union[str, File]] = None
-    preprocessor: EPreProcessor = EPreProcessor.canny
-    guide_image_unprocessed: Optional[Union[str, File]] = None
-
-    def __post_init__(self):
-        # This method will run after the default __init__ generated by @dataclass
-        guide_images = [self.guide_image, self.guide_image_unprocessed]
-        # Check that exactly one of the fields is provided
-        if sum(x is not None for x in guide_images) != 1:
-            raise ValueError(
-                "Exactly one of 'guide_image' or 'guide_image_unprocessed' must be provided, not both or none."
-            )
+class IControlNetAWithUUID(IControlNetGeneralWithUUID):
+    preprocessor: EPreProcessor
 
 
 @dataclass
-class IControlNetAWithUUID(IControlNetBaseWithUUID):
-    guide_image_uuid: str
-    weight: float
-    start_step: int
-    end_step: int
-    control_mode: EControlMode
-    preprocessor: EPreProcessor
-    guide_image: Optional[Union[str, File]] = None
-    guide_image_unprocessed: Optional[Union[str, File]] = None
-
-    def __post_init__(self):
-        # This method will run after the default __init__ generated by @dataclass
-        guide_images = [self.guide_image, self.guide_image_unprocessed]
-        # Check that exactly one of the fields is provided
-        if sum(x is not None for x in guide_images) != 1:
-            raise ValueError(
-                "Exactly one of 'guide_image' or 'guide_image_unprocessed' must be provided, not both or none."
-            )
+class IControlNetCannyWithUUID(IControlNetGeneralWithUUID):
+    lowThresholdCanny: int
+    highThresholdCanny: int
+    preprocessor: EPreProcessor = EPreProcessor.canny
 
 
 @dataclass
-class IControlNetHandsAndFaceWithUUID(IControlNetBaseWithUUID):
-    guide_image_uuid: str
-    weight: float
-    start_step: int
-    end_step: int
-    control_mode: EControlMode
+class IControlNetHandsAndFaceWithUUID(IControlNetGeneralWithUUID):
     preprocessor: EOpenPosePreProcessor
-    include_hands_and_face_open_pose: bool
-    guide_image_uuid: str
-    guide_image: Optional[Union[str, File]] = None
-    guide_image_unprocessed: Optional[Union[str, File]] = None
-
-    def __post_init__(self):
-        # This method will run after the default __init__ generated by @dataclass
-        guide_images = [self.guide_image, self.guide_image_unprocessed]
-        # Check that exactly one of the fields is provided
-        if sum(x is not None for x in guide_images) != 1:
-            raise ValueError(
-                "Exactly one of 'guide_image' or 'guide_image_unprocessed' must be provided, not both or none."
-            )
+    includeHandsAndFaceOpenPose: bool
```

Comment:
I guess it would harm backward comp too

Reply:
controlnet was broken (not working), so I don't think it will break backwards comp at this point